core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -358,14 +358,16 @@ private[neotypes] object DeferredQuery {
   * }}}
   */
 final class DeferredQueryBuilder private[neotypes] (private val parts: List[DeferredQueryBuilder.Part]) {
-  import DeferredQueryBuilder.{PARAMETER_NAME_PREFIX, Param, Part, Query}
+  import DeferredQueryBuilder.{PARAMETER_NAME_PREFIX, Param, Part, Query, SubQueryParam}
 
   /** Creates a [DeferredQuery] from this builder.
     *
     * @tparam T the result type of the constructed query.
     * @return A [DeferredQuery[T]]
     */
-  def query[T]: DeferredQuery[T] = {
+  def query[T]: DeferredQuery[T] = query(PARAMETER_NAME_PREFIX)
+
+  private[neotypes] def query[T](parameterNamePrefix: String): DeferredQuery[T] = {
```

Comment:
I do not understand the purpose of this change?

Reply:
The parameterized version is used by `CypherStringInterpolator.subQueryParts` to prefix subquery parameters so they don't collide...

But thinking about it made me realize that this will probably not work if there is more than one level of nested subqueries :/

Reply:
That is a good test actually!